nifi-extension-bundles/nifi-network-bundle/nifi-network-processors/src/main/java/org/apache/nifi/processors/network/SplitPCAP.java

```diff
@@ -0,0 +1,203 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.processors.network;
+
+import org.apache.nifi.components.PropertyDescriptor;
+import org.apache.nifi.flowfile.FlowFile;
+import org.apache.nifi.annotation.lifecycle.OnScheduled;
+import org.apache.nifi.annotation.behavior.WritesAttribute;
+import org.apache.nifi.annotation.documentation.CapabilityDescription;
+import org.apache.nifi.annotation.documentation.Tags;
+import org.apache.nifi.processor.AbstractProcessor;
+import org.apache.nifi.processor.ProcessContext;
+import org.apache.nifi.processor.ProcessSession;
+import org.apache.nifi.processor.Relationship;
+import org.apache.nifi.processor.util.StandardValidators;
+import org.apache.nifi.processors.network.util.PCAP;
+import org.apache.nifi.processors.network.util.PCAP.ByteBufferInterface;
+import org.apache.nifi.processors.network.util.PCAP.Packet;
+
+import java.io.ByteArrayOutputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+@Tags({"PCAP", "Splitter", "Network", "Packet", "Capture", "Wireshark", "TShark"})
+@CapabilityDescription("Splits a pcap file into multiple pcap files based on a maximum size.")
+@WritesAttribute(attribute = "ERROR_REASON", description = "The reason the flowfile was sent to the failure relationship.")
+
+public class SplitPCAP extends AbstractProcessor {
+
```

Comment:
`"ERROR_REASON"` is specified four times. Place it in a static variable.
```suggestion
@WritesAttribute(attribute = SplitPCAP.ERROR_REASON, description = "The reason the flowfile was sent to the failure relationship.")

public class SplitPCAP extends AbstractProcessor {

private static final String ERROR_REASON = "ERROR_REASON";
```

Reply:
I had to make this public or it wasn't visible from line 41

Reply:
Sorry about that. I am not sure `public `is needed. I believe package visibility would work also.

Reply:
I've changed it to `protected` and that seems to work.